### PR TITLE
add support for libjpeg-turbo and deprecate jpeg option

### PR DESCRIPTION
**Pull request by Renari**

### CODE REVIEW

**Comment by Renari** [comment]
It looks like this build is failing due to linking issues with openjp2 and openexr.

File: conanfile.py [hunk 1]
```diff
@@ -29,7 +29,8 @@ class ImageMagicConan(ConanFile):
                "webp": [True, False],
                "xml": [True, False],
                "freetype": [True, False],
-               "utilities": [True, False]}
+               "utilities": [True, False],
+               "with_libjpeg": [True, False, "libjpeg-turbo"]}
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
               "with_libjpeg": [False, "libjpeg", "libjpeg-turbo"]}
```

We had a similar discussion on Conan Center Index few days ago [here](https://github.com/conan-io/conan-center-index/pull/2730).

**Review by uilianries — CHANGES_REQUESTED** [review]

File: conanfile.py [hunk 2]
```diff
@@ -44,6 +45,7 @@ class ImageMagicConan(ConanFile):
                        "png": True,
                        "tiff": True,
                        "webp": True,
+                       "with_libjpeg": True,
```
Comment (uilianries) [inline conanfile.py]:
```suggestion
                       "with_libjpeg": "libjpeg",
```

**Review by uilianries — COMMENTED** [review]
small fix required

**Comment by Renari** [comment]
```
        # Handle deprecated jpeg option
        if self.options.jpeg and not self.options.with_libjpeg:
            self.output.warn("jpeg option is deprecated, use with_libjpeg option instead.")
            self.options.with_libjpeg = True
        del self.options.jpeg
```
This no longer works with the above changes due to `with_libjpeg` now having a default option. I think we can change the default for the `jpeg` option to `False`, then check if it's set to `True` and show the warning then since that means the option was being set by the user.

**Comment by Croydon** [comment]
This recipe is now in the [Conan Center Index](https://github.com/conan-io/conan-center-index/).

Please migrate to the recipe version there. 
If this is still an issue, please create a new issue in the [Conan Center Index issue tracker](https://github.com/conan-io/conan-center-index/issues).

Thanks!

https://conan.io/center/imagemagick

**Pull request closed without merging**